test/codegen.jl

```diff
@@ -49,36 +49,32 @@ end
 # Have to go through pains with recursive function (eval probably not required) to make sure
 # that inlining won't happen.
 function test_jl_dump_compiles()
-    tfile = tempname()
-    io = open(tfile, "w")
-    ccall(:jl_dump_compiles, Void, (Ptr{Void},), io.handle)
-    eval(@noinline function test_jl_dump_compiles_internal(x)
-        if x > 0
-            test_jl_dump_compiles_internal(x-1)
-        end
-        end)
-    test_jl_dump_compiles_internal(1)
-    ccall(:jl_dump_compiles, Void, (Ptr{Void},), C_NULL)
-    close(io)
-    tstats = stat(tfile)
-    tempty = tstats.size == 0
-    rm(tfile)
-    @test tempty == false
+    mktemp() do tfile, io
+        ccall(:jl_dump_compiles, Void, (Ptr{Void},), io.handle)
+        eval(@noinline function test_jl_dump_compiles_internal(x)
+            if x > 0
+                test_jl_dump_compiles_internal(x-1)
+            end
+            end)
+        test_jl_dump_compiles_internal(1)
+        ccall(:jl_dump_compiles, Void, (Ptr{Void},), C_NULL)
+
+        @test stat(io).size == 0
```

Comment:
Hmm you think? Maybe I'm missing something, but it seemed to me the intention is to test what gets written to the `io` handle that we pass to `:jl_dump_compiles` -- namely, test that nothing gets written (and/or (unlikely but possible) that everything gets truncated after writing). In any case, shouldn't that be the same between calling `fstat` on the file handle and calling `stat` on the file name?